packages/twenty-front/src/modules/object-record/record-show/hooks/useRecordShowPagePagination.ts

```diff
@@ -0,0 +1,224 @@
+  const [objectRecords, setObjectRecords] = useState<ObjectRecord[]>([]);
+  const [totalRecords, setTotalRecords] = useState(0);
+  const [isLoadedRecords, setIsLoadedRecords] = useState(false);
+  const [isLoadingPagination, setIsLoadingPagination] = useState(true);
+  const [hasNextPage, setHasNextPage] = useState<boolean | undefined>();
+
+  const [viewName, setViewName] = useState('');
+
+  const objectNameSingular = propsObjectNameSingular || paramObjectNameSingular;
+  const objectRecordId = propsObjectRecordId || paramObjectRecordId;
+
+  if (!objectNameSingular || !objectRecordId) {
+    throw new Error('Object name or Record id is not defined');
+  }
+
+  const { objectMetadataItem } = useObjectMetadataItem({ objectNameSingular });
+
+  const { objectNamePlural } = useObjectNamePluralFromSingular({
+    objectNameSingular,
+  });
+
+  const { records: views } = usePrefetchedData(PrefetchKey.AllViews);
+
+  const view = useMemo(() => {
+    if (!viewIdQueryParam) {
+      return views.find(
+        (view: any) =>
+          view.key === 'INDEX' &&
+          view?.objectMetadataId === objectMetadataItem.id,
+      ) as View;
+    } else {
+      return views.find(
+        (view: any) =>
+          view?.id === viewIdQueryParam &&
+          view?.objectMetadataId === objectMetadataItem.id,
+      ) as View;
+    }
+  }, [viewIdQueryParam, objectMetadataItem, views]);
+
+  const activeFieldMetadataItems = useMemo(
+    () =>
+      objectMetadataItem
+        ? objectMetadataItem.fields.filter(
+            ({ isActive, isSystem }) => isActive && !isSystem,
+          )
+        : [],
+    [objectMetadataItem],
+  );
+
+  const filterDefinitions = formatFieldMetadataItemsAsFilterDefinitions({
+    fields: activeFieldMetadataItems,
+  });
+
+  const sortDefinitions = formatFieldMetadataItemsAsSortDefinitions({
+    fields: activeFieldMetadataItems,
+  });
+
+  const filter = turnObjectDropdownFilterIntoQueryFilter(
+    mapViewFiltersToFilters(view?.viewFilters ?? [], filterDefinitions),
+    objectMetadataItem?.fields ?? [],
+  );
+
+  const orderBy = turnSortsIntoOrderBy(
+    objectMetadataItem,
+    mapViewSortsToSorts(view?.viewSorts ?? [], sortDefinitions),
+  );
+
+  const { fetchMoreRecords } = useFindManyRecords({
+    filter,
+    orderBy,
+    skip: isLoadedRecords,
+    objectNameSingular,
+    recordGqlFields: generateDepthOneRecordGqlFields({
+      objectMetadataItem,
+    }),
+    onCompleted: (records, options) => {
+      setObjectRecords(records);
+      setTotalRecords(options?.totalCount ?? 0);
+      setHasNextPage(options?.pageInfo?.hasNextPage);
+      setIsLoadingPagination(false);
+    },
+  });
+
+  useEffect(() => {
+    if (
+      !isLoadingPagination &&
+      objectRecords.length > 0 &&
+      isDefined(hasNextPage)
+    ) {
+      const recordIndex = objectRecords.findIndex(
+        (rec) => rec.id === objectRecordId,
+      );
+      if (recordIndex < 0 && hasNextPage) {
+        fetchMoreRecords();
```

Comment:
Need to change the mechanism to do two queries for previous/past results ; or maybe even better if it works, one query with an offset (can be adapted based on position) to get in a single query roughly half of the records before and half of the records after

Reply:
Note one thing we could have done to get the exact position in the list without passing it as a state from the previous state, would have been to convert the sort criteria into an additional (e.g. if it's sortBy createdAt, then we add createdAt < currentRecord.createdAt to the current record filters). It would have been perfect, but it might be a bit too painful to implement because "greater/less than" filters are probably not implemented for all field types on the backend.